.github/workflows/ci-cd.yml

```diff
@@ -127,12 +124,36 @@ jobs:
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
 
+      - name: Store the distribution packages
+        uses: actions/upload-artifact@v3
+        with:
+          name: python-package-distributions
+          path: dist/
+
+  release:
+    needs:
+      - build_release
+
+    runs-on: ubuntu-latest
+    environment: pypi
+    concurrency: release
+    permissions:
+      id-token: write
+      contents: write
+
+    steps:
+      - name: Download all the dists
+        uses: actions/download-artifact@v3
+        with:
+          name: python-package-distributions
+          path: dist/
+
       - name: Publish package distributions to PyPI
         uses: pypa/gh-action-pypi-publish@release/v1
-        if: steps.release.outputs.released == 'true'
+        if: steps.build_release.outputs.released == 'true'
 
       - name: Publish package distributions to GitHub Releases
         uses: python-semantic-release/upload-to-gh-release@main
-        if: steps.release.outputs.released == 'true'
+        if: steps.build_release.outputs.released == 'true'
```

Comment:
I think this change will work, but I'm not sure how isolated `environment`s are so it might break

Reply:
No, this won't work. The `${{ steps }}` context refers to the steps within a job. If you want to expose this from the build job, you'd have to declare `outputs` on that job, wiring the step outputs to job outputs, and then, use a `${{ needs }}` context here, to access the values.

Reply:
Thanks. 

I _think_ I've fixed this in eae00f4539c49c02e960121c14f633afad16c778